src/Components/Promises/PromiseTable/index.js

```diff
@@ -11,50 +11,87 @@ import {
 import InCompleteIcon from 'material-ui/svg-icons/navigation/close';
 import CompleteIcon from 'material-ui/svg-icons/navigation/check';
 import { red500, green500 } from 'material-ui/styles/colors';
-import type { PromiseType } from '../../../types';
+import PromiseTableAdminPanel from '../PromiseTableAdminPanel';
+import type { PromiseType, PromiseId, SubjectId } from '../../../types';
 
 export type Props = {
   promises: Array<PromiseType>,
+  linkSubjectToPromises: (promiseIds: Array<PromiseId>, subjectId: SubjectId) => void,
 };
 
 const getIcon = fulfilled =>
   fulfilled ? <CompleteIcon color={green500} /> : <InCompleteIcon color={red500} />;
 
-const PromiseTable = (props: Props) => {
-  const { promises } = props;
-  if (!promises) {
-    return null;
-  }
+export class PromiseTable extends React.Component {
+  props: Props;
 
-  const promiseRows = promises.map(promise =>
-    <TableRow key={promise.id}>
-      <TableRowColumn>{promise.name}</TableRowColumn>
-      <TableRowColumn>{promise.politician}</TableRowColumn>
-      <TableRowColumn>{promise.small_description}</TableRowColumn>
-      <TableRowColumn>{getIcon(promise.fulfilled)}</TableRowColumn>
-    </TableRow>
-  );
-
-  return (
-    <Table fixedHeader={true} multiSelectable={true}>
-      <TableHeader>
-        <TableRow>
-          <TableHeaderColumn colSpan="3">
-            Here be buttons
-          </TableHeaderColumn>
-        </TableRow>
-        <TableRow>
-          <TableHeaderColumn>Name</TableHeaderColumn>
-          <TableHeaderColumn>Politician id</TableHeaderColumn>
-          <TableHeaderColumn>Short description</TableHeaderColumn>
-          <TableHeaderColumn>Fulfilled</TableHeaderColumn>
-        </TableRow>
-      </TableHeader>
-      <TableBody>
-        {promiseRows}
-      </TableBody>
-    </Table>
-  );
-};
+  state = {
+    selectedPromises: [],
+  };
+
+  linkSubjectsCallback = (subjectId: SubjectId) => {
+    const { linkSubjectToPromises } = this.props;
+    const { selectedPromises } = this.state;
+
+    const promiseIds = selectedPromises.map(promise => promise.id);
+    linkSubjectToPromises(promiseIds, subjectId);
+  };
+
+  onRowSelect = (selection: 'all' | 'none' | Array<number>) => {
+    const refine = selection;
+    if (refine === 'all') {
+      this.setState({
+        selectedPromises: this.props.promises,
+      });
+    } else if (refine === 'none') {
+      this.setState({
+        selectedPromises: [],
+      });
+    } else if (Array.isArray(refine)) {
+      const proms = this.props.promises.filter((p, idx) => refine.includes(idx));
+      this.setState({ selectedPromises: proms });
+    }
+  };
+
+  render() {
+    const { promises } = this.props;
+    if (!promises) {
+      return null;
```

Comment:
Add a loading bar here?

Reply:
Good idea.

Would you agree that there should be a `loadingBar` PR upcoming soon?
As opposed to bloating this PR any more

Reply:
yeah probably some other places that could use a loading bar